### PR TITLE
NEW sliderWidget

### PR DESCRIPTION
**Pull request by radjkarl**
image
image

### CODE REVIEW

File: pyqtgraph/widgets/SliderWidget.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        #number of floating points:
+        self.precission = int(round( np.log10( (self.step / (self.mx-self.mn) )) ) )
+        if self.precission < 0: 
+            self.precission = 0
+        #length of the number in the label:
+        self.valueLen = max(len(str(int(self.mn))), len(str(int(self.mx))) ) + self.precission
+
+    def setOpts(self, bounds=None):
+        if bounds != None:
+            self.setRange(bounds)
```
Comment (miili) [inline pyqtgraph/widgets/SliderWidget.py]: make this `self.setRange(*bounds)`

Comment (radjkarl) [inline pyqtgraph/widgets/SliderWidget.py]: thanks. Done.

**Comment by miili** [comment]
Respecting for `opts['suffix']` would be great! :wink: 

thanks for the widget :+1:

M

**Comment by radjkarl** [comment]
added support for opts['suffix']
image

**Comment by j9ac9k** [comment]
CI failure across all pipelines

Specifically `examples/test_examples.py::testExamples[ parametertree.py - <qt-binding> ]`

**Comment by ntjess** [comment]
👋 @radjkarl  these options should now be supported by #1844 via c3802ec. Feel free to take a look! I will close this as a result, since that one is significantly easier to merge into the current master branch, but I'm more than happy to continue the conversation here if desired.

**Pull request closed without merging**